src/fenicsxconcrete/problem.py

```diff
@@ -0,0 +1,73 @@
+from pydantic.dataclasses import dataclass
```

Comment:
The `problem.py` is more an example for trying out how a full json would look like. 

I personally see benefits for storing everything that I added because in a workflow, I would could then edit a json that is read by the simulation script instead of changing the script. 

As for what should always be contained in a problem, I would say that it should always be able to return sensor readings and it should always be able to plot all variables that are stored. The input can be an experiment and a dictionary of parameters. But apart from that, I am fine with not restricting what a problem can do

Reply:
I'm wondering what the advantage of the plotting capabilities in the problem is, instead of just adding that (and that is something everyone could decide himself) when creating the model. Then this class can be used afterwards to do the plotting (and does not even has to know the FE-problem), just the mesh, which is provided in the experiment. Then you can use any tool to visualize your results, potentially even in a postprocessing step (when having serialized the results) with a different software tool (visualizing ansys results with pyvista)